stm32: add more target boards to dac testcase

FRASTM opened the pull request:
This adds the configuration to run tests/drivers/dac/dac_api
on the nucleo_g071rb and nucleo_g474re,  the nucleo_wl55jc, nucleo_h743zi, disco_l475_iot1 nucleo_f091rc 
 nucleo_f207zg nucleo_f746zg

It adds the following boards to run the samples/drivers/dac application:
nucleo_g474re,  the nucleo_wl55jc, nucleo_h743zi, disco_l475_iot1 nucleo_f207zg b_u585i_iot02a nucleo_f746zg

DAC1 out1 pin is set by default

Signed-off-by: Francois Ramu [francois.ramu@st.com](mailto:francois.ramu@st.com)

FRASTM committed:
boards: stm32 adding dac features on nucleo boards nucleo_wl55jc

It enables he DAC peripheral on the nucleo_wl55jc,
target boards from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: stm32 adding dac features on nucleo boards nucleo_h743zi

It enables he DAC peripheral on the nucleo_h743zi target
boards from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
tests: drivers: DAC testcase running on more stm32 nucleo boards

It includes the testcase for existing nucleo target boards.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: stm32 adding dac features on nucleo boards nucleo_g474re

It enables he DAC peripheral on the nucleo_g474re target
boards from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
samples: drivers: DAC testcase running on more stm32 boards

It includes the samples/drivers/dac application for
existing nucleo and disco target boards.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

martinjaeger commented on boards/arm/nucleo_g474re/nucleo_g474re.dts:
```diff
@@ -173,6 +173,12 @@
 };
 
+&dac1 {
+	pinctrl-0 = <&dac1_out1_pa4>;
```
The pin PA4 should also be mentioned in the docs (for all new boards in this PR).

FRASTM replied:
dac1_out1_pa4 added to the doc of the nucleo_g474re

martinjaeger commented on samples/drivers/dac/sample.yaml:
```diff
@@ -4,10 +4,11 @@ tests:
     tags: DAC
     platform_allow: |
-      arduino_zero frdm_k22f frdm_k64f nucleo_f091rc nucleo_g071rb nucleo_g431rb
+      arduino_zero frdm_k22f frdm_k64f nucleo_f091rc nucleo_g071rb nucleo_g474re
```
Out of curiosity: Why is the `nucleo_g431rb` removed? To reduce number of test runs for same SOC family? Is there any advantage to test with `g474` instead of `ge431`?

FRASTM replied:
We had a nucleo_g474re board available for testing, so it was choosen to represent the stm32g4 soc serie. The nucleo_g431re is still a valid target. Restoring

FRASTM replied:
Test results:
```
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_g474re,passed,,na,3.625081777572632,0,0
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_wl55jc,passed,,na,6.49380350112915,0,0
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_h743zi,passed,,na,3.405778169631958,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_g474re,passed,,na,2.5104336738586426,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_wl55jc,passed,,na,4.01857590675354,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_h743zi,passed,,na,2.741598606109619,0,0
```

martinjaeger commented on tests/drivers/dac/dac_api/testcase.yaml:
```diff
@@ -5,5 +5,6 @@ tests:
       frdm_k22f frdm_k64f nucleo_l073rz nucleo_l152re twr_ke18f nucleo_f767zi
       nucleo_f429zi bl652_dvk bl653_dvk bl654_dvk bl5340_dvk_cpuapp stm32f3_disco
-      stm32l562e_dk nucleo_l552ze_q b_u585i_iot02a
+      stm32l562e_dk nucleo_l552ze_q b_u585i_iot02a nucleo_g474re nucleo_g071rb
+      nucleo_wl55jc nucleo_h743zi nucleo_l476rg nucleo_f091rc nucleo_f207zg
```
It would be nice to add new boards in alphabetical order, same as the defines.

FRASTM replied:
changing to `nucleo_f207zg nucleo_g474re nucleo_h743zi nucleo_l476rg nucleo_wl55jc`

martinjaeger replied:
Still neither alphabetical order nor same order as in the `#define`s in `tests/drivers/dac/dac_api/src/test_dac.c`...

FRASTM commented:
Test results:
```
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_g071rb,passed,,na,3.214419364929199,0,0
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_g474re,passed,,na,3.625081777572632,0,0
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_wl55jc,passed,,na,6.49380350112915,0,0
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_h743zi,passed,,na,3.405778169631958,0,0
tests/drivers/dac/dac_api/drivers.dac,arm,nucleo_f091rc,passed,,na,2.912710666656494,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_g071rb,passed,,na,2.9022881984710693,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_g474re,passed,,na,2.5104336738586426,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_wl55jc,passed,,na,4.01857590675354,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_h743zi,passed,,na,2.741598606109619,0,0
samples/drivers/dac/sample.drivers.dac,arm,nucleo_f091rc,passed,,na,7.266473293304443,0,0
```

martinjaeger commented on boards/arm/nucleo_h743zi/nucleo_h743zi.dts:
```diff
@@ -127,6 +127,12 @@ zephyr_udc0: &usbotg_fs {
 
+&dac1 {
+	status = "okay";
+	pinctrl-0 = <&dac1_out1_pa4>;
```
Please add chosen pin to docs also for this board.

FRASTM replied:
done

martinjaeger commented on boards/arm/nucleo_l476rg/nucleo_l476rg.dts:
```diff
@@ -173,3 +173,9 @@
 };
+
+&dac1 {
+	pinctrl-0 = <&dac1_out1_pa4>;
```
Please add chosen pin to docs also for this board.

FRASTM replied:
removed, not used anymore

martinjaeger commented on boards/arm/nucleo_wl55jc/nucleo_wl55jc.dts:
```diff
@@ -135,6 +135,13 @@
 
+
+&dac1 {
+	pinctrl-0 = <&dac_out1_pa10>;
```
Please add chosen pin to docs also for this board.

FRASTM replied:
done

FRASTM commented:
replace nucleo_l476rg with nucleo_l4r5zi as this board is available

FRASTM committed:
boards: stm32 adding dac features on the disco kit stm32l475

It enables he DAC peripheral on the disco_l475_iot1 target
boards from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>

martinjaeger commented on boards/arm/nucleo_wl55jc/doc/nucleo_wl55jc.rst:
```diff
@@ -234,6 +236,7 @@ Default Zephyr Peripheral Mapping:
 - SPI_1_MISO : PA6 (arduino_spi)
 - SPI_1_MOSI : PA7 (arduino_spi)
 - ADC1_IN5 : PB1 (Arduino pin A0)
+- DAC1OUT1 : PA10 (Arduino pin A2)
```
```suggestion
- DAC1_OUT1 : PA10 (Arduino pin A2)
```

FRASTM replied:
changed

FRASTM committed:
boards: stm32 adding dac features on nucleo boards nucleo_f746zg

It enables he DAC peripheral on the nucleo_f746zg target
boards from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>
